test_integration/base_suite_test.go

```diff
@@ -0,0 +1,285 @@
+	err := deleteDockerTestImagesAndPrune(s.suiteRunIdentifier)
+	s.Require().NoError(err, "Failed to delete images")
+}
+
+func (s *BaseDockerComposeTestSuite) SetupTest() {
+	// Runs before each test
+	s.T().Log("Setting up [Test] in base suite...")
+}
+
+func (s *BaseDockerComposeTestSuite) TearDownTest() {
+	// Runs after each test
+	s.T().Log("Tearing down [Test] in base suite......")
+}
+
+func (s *BaseDockerComposeTestSuite) copyTestsAssetsToJumpbox() {
+	s.T().Log("Copying ./assets to jumpbox container......")
+
+	// TODO: make this name configurable
+	jumpboxServiceName := "bacalhau-jumpbox-node"
+	typeAssertedComposeStack, typeAssertionOk := s.ComposeStack.(interface {
+		ServiceContainer(context.Context, string) (*testcontainers.DockerContainer, error)
+	})
+	s.Require().True(typeAssertionOk, "copying test assets: ComposeStack does not implement ServiceContainer method")
+
+	jumpBoxContainer, err := typeAssertedComposeStack.ServiceContainer(s.Context, jumpboxServiceName)
+	s.Require().NoErrorf(err, "cannot find jumpbox container: %q", err)
+
+	testAssetsDirectory, err := filepath.Abs(filepath.Join(".", "assets/"))
+	s.Require().NoErrorf(err, "cannot find assets tes directory: %q", err)
+
+	// Now you can use CopyDirToContainer on the service container
+	err = jumpBoxContainer.CopyDirToContainer(s.Context, testAssetsDirectory, "/app/", 0o755)
+	s.Require().NoErrorf(err, "cannot copy tests assets to jumpbox: %q", err)
+}
+
+func (s *BaseDockerComposeTestSuite) standUpDockerComposeStack(composeFilePath string) {
+	s.Require().FileExistsf(composeFilePath, "Rendered Docker compose File does not exist: %s", composeFilePath)
+
+	// Testify project names must consist only of lowercase alphanumeric characters, hyphens, and underscores
+	normalizedTestName := strings.ToLower(strings.ReplaceAll(s.T().Name(), "/", "-"))
+	identifier := fmt.Sprintf("test-bacalhau-%s-%s", s.suiteRunIdentifier, normalizedTestName)
+
+	composeStack, err := tc.NewDockerComposeWith(
+		tc.StackIdentifier(identifier),
+		tc.WithStackFiles(composeFilePath),
+	)
+	s.Require().NoErrorf(err, "Error preparing Docker Compose Stack: %q\n", err)
+
+	s.ComposeStack = composeStack
+
+	err = composeStack.Up(s.Context, tc.Wait(true), tc.WithRecreate(api.RecreateForce))
+	s.Require().NoErrorf(err, "Error creating Docker Compose Stack: %q\n", err)
+}
+
+func (s *BaseDockerComposeTestSuite) downWithDockerCompose() {
+	err := s.ComposeStack.(interface {
+		Down(context.Context, ...tc.StackDownOption) error
+	}).Down(
+		context.Background(),
+		tc.RemoveImagesLocal,
+		tc.RemoveOrphans(true),
+		tc.RemoveVolumes(true),
+	)
+	s.Require().NoErrorf(err, "Error tearing down docker compose stack: %q\n", err)
+	s.Cancel()
+}
+
+func (s *BaseDockerComposeTestSuite) executeCommandInContainer(containerName string, cmd []string) (string, error) {
+	typeAssertedComposeStack, ok := s.ComposeStack.(interface {
+		ServiceContainer(context.Context, string) (*testcontainers.DockerContainer, error)
+	})
```

Comment:
Since this type assertion is used in several places, do we think it's worth defining it as an interface definition, rather than inline?

Reply:
Good point, I will be doing that in my current task related to Bashtub switch.